src/audio/igo_nr/igo_nr.c

```diff
@@ -0,0 +1,521 @@
+	if (ret < 0) {
+		comp_cl_err(&comp_igo_nr, "igo_nr_new(): comp_init_data_blob() failed.");
+		goto err;
+	}
+	dev->state = COMP_STATE_READY;
+
+	comp_cl_info(&comp_igo_nr, "igo_nr created");
+
+	return dev;
+
+err:
+	rfree(dev);
+	rfree(cd);
+	return NULL;
+}
+
+static void igo_nr_free(struct comp_dev *dev)
+{
+	struct igo_nr_comp_data *cd = comp_get_drvdata(dev);
+
+	comp_info(dev, "igo_nr_free()");
+
+	comp_data_blob_handler_free(cd->model_handler);
+
+	rfree(cd);
+	rfree(dev);
+}
+
+static int32_t igo_nr_cmd_get_data(struct comp_dev *dev,
+				   struct sof_ipc_ctrl_data *cdata,
+				   int32_t max_size)
+{
+	struct igo_nr_comp_data *cd = comp_get_drvdata(dev);
+	int32_t ret = 0;
+
+	switch (cdata->cmd) {
+	case SOF_CTRL_CMD_BINARY:
+		comp_info(dev, "igo_nr_cmd_get_data(), SOF_CTRL_CMD_BINARY");
+		ret = comp_data_blob_get_cmd(cd->model_handler, cdata, max_size);
+		break;
+	default:
+		comp_err(dev, "igo_nr_cmd_get_data() error: invalid cdata->cmd");
+		ret = -EINVAL;
+		break;
+	}
+	return ret;
+}
+
+static int32_t igo_nr_cmd_set_data(struct comp_dev *dev,
+				   struct sof_ipc_ctrl_data *cdata)
+{
+	struct igo_nr_comp_data *cd = comp_get_drvdata(dev);
+	int32_t ret = 0;
+
+	switch (cdata->cmd) {
+	case SOF_CTRL_CMD_BINARY:
+		comp_info(dev, "igo_nr_cmd_set_data(), SOF_CTRL_CMD_BINARY");
+		ret = comp_data_blob_set_cmd(cd->model_handler, cdata);
+		break;
+	default:
+		comp_err(dev, "igo_nr_cmd_set_data() error: invalid cdata->cmd");
+		ret = -EINVAL;
+		break;
+	}
+
+	return ret;
+}
+
+/* used to pass standard and bespoke commands (with data) to component */
+static int32_t igo_nr_cmd(struct comp_dev *dev,
+			  int32_t cmd,
+			  void *data,
+			  int32_t max_data_size)
+{
+	struct sof_ipc_ctrl_data *cdata = data;
+	int32_t ret = 0;
+
+	comp_info(dev, "igo_nr_cmd()");
+
+	switch (cmd) {
+	case COMP_CMD_SET_DATA:
+		ret = igo_nr_cmd_set_data(dev, cdata);
+		break;
+	case COMP_CMD_GET_DATA:
+		ret = igo_nr_cmd_get_data(dev, cdata, max_data_size);
+		break;
+	default:
+		comp_err(dev, "igo_nr_cmd() error: invalid command");
+		ret = -EINVAL;
+	}
+
+	return ret;
+}
+
+static void igo_nr_process(struct comp_dev *dev,
+			   struct comp_buffer *source,
+			   struct comp_buffer *sink,
+			   int32_t src_frames,
+			   int32_t sink_frames,
+			   uint32_t source_bytes,
+			   uint32_t sink_bytes)
+{
+	struct igo_nr_comp_data *cd = comp_get_drvdata(dev);
+
+	buffer_invalidate(source, source_bytes);
+
+	cd->igo_nr_func(cd, &source->stream, &sink->stream, src_frames, sink_frames);
+
+	buffer_writeback(sink, sink_bytes);
+
+	/* calc new free and available */
+	comp_update_buffer_consume(source, source_bytes);
+	comp_update_buffer_produce(sink, sink_bytes);
+}
+
+static void igo_nr_get_igo_params(struct comp_dev *dev)
+{
+	struct igo_nr_comp_data *cd = comp_get_drvdata(dev);
+
+	comp_info(dev, "igo_nr_get_igo_params()");
+
+	struct sof_igo_nr_config *p_config = comp_get_data_blob(cd->model_handler, NULL, NULL);
```

Comment:
p_config should be declared before the comp_info()

Reply:
fixed. thanks!